src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs

```diff
@@ -258,9 +294,51 @@ private List<string> AddArgs()
                 allArgs.Add("--logger:Console;Verbosity=" + vsTestVerbosity);
             }
 
-            if (!string.IsNullOrEmpty(this.VSTestBlame))
+            var blameCrash = !string.IsNullOrEmpty(this.VSTestBlameCrash);
+            var blameHang = !string.IsNullOrEmpty(this.VSTestBlameHang);
+            if (!string.IsNullOrEmpty(this.VSTestBlame) || blameCrash || blameHang)
             {
-                allArgs.Add("--Blame");
+                var blameArgs = "--Blame";
+
+                var dumpArgs = new List<string>();
+                if (blameCrash || blameHang)
+                {
+                    if (blameCrash)
+                    {
+                        dumpArgs.Add("CollectDump");
+                        if (!string.IsNullOrEmpty(this.VSTestBlameCrashCollectAlways))
+                        {
+                            dumpArgs.Add($"CollectAlways={this.VSTestBlameCrashCollectAlways}");
+                        }
+
+                        if (!string.IsNullOrEmpty(this.VSTestBlameCrashDumpType))
+                        {
+                            dumpArgs.Add($"DumpType={this.VSTestBlameCrashDumpType.Trim()}");
```

Comment:
Why do use Trim() only here, not in other adds to List, for instance in above ` dumpArgs.Add($"CollectAlways={this.VSTestBlameCrashCollectAlways}");`?

Reply:
Don't remember, fixing.